board.rb

```diff
@@ -15,45 +18,93 @@ def graphic_table
     counter = 0
     element = division = ""
 
-    for i in (0...@size)
-      for j in (0...@size)
-      
-        if @table[i][j].nil?
-          element += '-' + '|'
-        else
-          element += @table[i][j].to_s + '|'
-        end
-        counter += 1
-      end
+    @table.each do |rows|
+      element += rows.map{ |val|  val.nil? ? '-|' : "#{val}|" }.join
       element += "\n"
-      division = '-' * (@size * 2) + "\n" if (counter % @size).zero?
+      division = "#{'-' * (@size * 2)} \n" if (counter % @size).zero?
       element += division
     end
 
     element
   end
 
   def fill_table(position, symbol)
-    arr_pos = position_coverter position
-    if empty_position? arr_pos
-      @table[arr_pos[0]][arr_pos[1]] = symbol
-      return true
-    else
-      return false
-    end
+    write_cell(position, symbol) if empty_position? position
   end
 
   def table_guide
     table_str = ""
     (1..@size**2).each do |num|
-      table_str += "#{num}" + ' ' * ((@size * @size).to_s.length - num.to_s.length) + '|'
+      table_str += "#{num}#{ ' ' * ((@size * @size).to_s.length - num.to_s.length) }|"
       if(num % @size).zero?
-        table_str += "\n" + '-' * ((@size * @size).to_s.length * @size + @size) + "\n"
+        table_str += "\n"
+        table_str += "#{'-' * ((@size * @size).to_s.length * @size + @size)} \n"
       end
     end
     table_str
    end
 
+   def check_rows
+    for row in @table
+      return WIN if row.uniq.size <= 1 && row.uniq != [nil]
+    end
+    ON_GAME
+  end
+
+  def check_columns
+    column_matrix = []
+
+    @size.times do |pos|
+      column_matrix << @table.map do |rows|
+        rows[pos]
+      end
+    end
+
+    for column in column_matrix
+      return WIN if column.uniq.size <= 1 && column.uniq != [nil]
+    end
+    ON_GAME
+  end
+
+  def check_main_diagonal
+    main_diagonal = []
+
+    main_diagonal = @table.map.with_index do |rows, i|
+      rows[i]
+    end
+
+    return WIN if main_diagonal.uniq.size <= 1 && main_diagonal.uniq != [nil]
+
+    ON_GAME
+  end
+
+  def check_second_diagonal
+    second_diagonal = []
+
+    second_diagonal = @table.map.with_index do |rows, i|
+      rows[@size - i - 1]
+    end
+
+    return WIN if second_diagonal.uniq.size <= 1 && second_diagonal.uniq != [nil]
+
+    ON_GAME
+  end
+
```

Comment:
@heyapricot about to method to run all the checks we think this is a responsibility of the game and we already have a method there (the `status` method) to check that.
About the line that is identical for all the check, we consider that we need it there because each method checks if there is win combination.

Reply:
Let's take a look at the `status` method inside `Game`
It calls the 4 methods that check the possible combinations. 
What I'm trying to say is that `Game` shouldn't even know or care about how the board determines the win (Calling each of `Board` methods for combination check)

Wouldn't it be nice to call something like `board.has_sequence?` and just get `true` or `false`?  

This `board.has_sequence?` can perform all the checks. `Game` only cares if there is a sequence, not which one

Let me know what you think. We can talk over here or on Crisp